### PR TITLE
#96 Refactored terraform

### PR DESCRIPTION
**Pull request by mahinuzzaman**
Feat: #96 
- I added the lambdas module to create lambda and its policies. so that we can just create modules with lambda options and the lambdas module will take care of the rest.
- Updated auth module to use lambda module
- Added Cloudwatch log group for API gateway
- Fixed file formatting issues

- [ ] Does not depend on other PRs
- [x] Depends on other PR being merged first: #98 

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [x] Improvement
- [ ] New feature
- [x] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?

Provide a brief overview of the changes introduced in this pull request.

## Related Issues/PRs (optional)

Link to the relevant issue(s) or provide a clear description of the problem or feature being addressed.

## How to test

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [ ] I have added or updated necessary documentation to reflect the changes.
- [ ] I have followed the project's coding style and guidelines.
- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [ ] I have checked for any potential conflicts with other branches or pull requests.
- [ ] I have updated necessary OpenAPI specs

## Reviewer Request (optional)

If you have any specific requests or suggestions for reviewers, mention them here.

## Screenshots (optional)

If applicable, provide any additional screenshots or GIFs that showcase the changes made in this pull request.

## Demo Link (optional)

If applicable, provide a link to a live demo or deployment where the changes can be reviewed.

## Known Issues (optional)

If there are any known issues or limitations with the changes made in this pull request, describe them here.

## Additional Notes (optional)

Add any additional notes or comments you think are relevant to this pull request.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

File: iac/terraform/aws/auth/lambdas.tf [hunk 1]
```diff
@@ -1,43 +1,49 @@
 locals {
+  lambda_archive_path = "${path.module}/../../../../core/services/aws/.build/zips"
   lambda_options = {
     refresh-token = {
       name     = "refresh-token"
-      zip_path = "${path.module}/../../../../core/services/aws/.build/zips/refreshToken.zip"
-      policy   = data.aws_iam_policy_document.lambda_common_policy
+      zip_path = "${local.lambda_archive_path}/refreshToken.zip"
       handler  = "refreshToken.default"
+      statement = [
```
Comment (melon-ruet) [inline iac/terraform/aws/auth/lambdas.tf:8]: This is not good. So multiple lambdas which will use same policy can not share policy.

**Review by melon-ruet — CHANGES_REQUESTED** [review]

File: iac/terraform/aws/auth/lambdas.tf (hunk 1 above)
Comment (mahinuzzaman) [reply iac/terraform/aws/auth/lambdas.tf:8]: every lambda will have its own statement block

Comment (melon-ruet) [reply iac/terraform/aws/auth/lambdas.tf:8]: Isn't sharing policy better with reusable terraform code?

Comment (mahinuzzaman) [reply iac/terraform/aws/auth/lambdas.tf:8]: yes, we can create local variables with policies. and we can use them in multiple places. this way we don't have to write duplicate codes

Comment (mahinuzzaman) [reply iac/terraform/aws/auth/lambdas.tf:8]: like this, its this looks ok
image

File: iac/terraform/aws/auth/lambdas.tf [hunk 2]
```diff
@@ -1,41 +1,21 @@
...
     refresh-token = {
-      name     = "refresh-token"
-      zip_path = "${path.module}/../../../../core/services/aws/.build/zips/refreshToken.zip"
-      policy   = data.aws_iam_policy_document.lambda_common_policy
-      handler  = "refreshToken.default"
+      name      = "refresh-token"
+      zip_path  = "${local.lambda_archive_path}/refreshToken.zip"
+      handler   = "refreshToken.default"
+      statement = local.policies.common_policies
       env_variables = {
```
Comment (melon-ruet) [inline iac/terraform/aws/auth/lambdas.tf:9]: Run `make run-command-tf-fmt` to format terraform

**Review by melon-ruet — CHANGES_REQUESTED** [review]

File: iac/terraform/aws/auth/lambdas.tf (hunk 2 above)
Comment (mahinuzzaman) [reply iac/terraform/aws/auth/lambdas.tf]: Updated in c7a8df7b774cad951c47e5d2a4ac76d8e394405c

**Review by melon-ruet — DISMISSED** [review]

**Review by melon-ruet — APPROVED** [review]

**Pull request merged**